syntax = "proto3";

package prod_im_server;
import "common.proto";

service prod_im_server_service {
    rpc user_register (user_register_req) returns (user_register_res) {}
    rpc login (login_req) returns (login_res) {}
    rpc get_contact_list (get_contact_list_req) returns (get_contact_list_res) {}
    rpc add_contact (add_contact_req) returns (add_contact_res) {}
    rpc del_contact (del_contact_req) returns (del_contact_res) {}
    rpc client_send_chat_msg (send_chat_msg_req) returns (send_chat_msg_res) {}
    rpc client_get_chat_msg (get_chat_msg_req) returns (get_chat_msg_res) {}
}

message user_register_req {
    string user_id = 1;
    string user_pass = 2;
}
message user_register_res {
    int32 result = 1;
}

message login_req {
    string user_id = 1;
    string user_pass = 2;
    uint32 client_port = 3;
}
message login_res {
    int32 result = 1;
}

message get_contact_list_req {
    string user_id = 1;
}
message get_contact_list_res {
    int32 result = 1;
    repeated common.prod_im_contact contact_list = 2;
}

message add_contact_req {
    string user_id = 1;
    string contact_id = 2;
    string contact_name = 3;
}
message add_contact_res {
    int32 result = 1;
}

message del_contact_req {
    string user_id = 1;
    string contact_id = 2;
}
message del_contact_res {
    int32 result = 1;
}

message send_chat_msg_req {
    string sender_id = 1;
    string receiver_id = 2;
    string chat_msg = 3;
}
message send_chat_msg_res {
    int32 result = 1;
}

message get_chat_msg_req {
    string user_id = 1;
    uint32 msg_index = 2;
}
message get_chat_msg_res {
    int32 result = 1;
    repeated common.prod_im_chat_msg msg_list = 2;
}
